\documentclass[journal, a4paper]{IEEEtran}

% some very useful LaTeX packages include:
\usepackage{cite}
\usepackage{plain}
\usepackage{cancel}
\renewcommand{\citedash}{--}    
%\usepackage{cite}      % Written by Donald Arseneau
                        % V1.6 and later of IEEEtran pre-defines the format
                        % of the cite.sty package \cite{} output to follow
                        % that of IEEE. Loading the cite package will
                        % result in citation numbers being automatically11
                        % sorted and properly "ranged". i.e.,
                        % [1], [9], [2], [7], [5], [6]
                        % (without using cite.sty)
                        % will become:
                        % [1], [2], [5]--[7], [9] (using cite.sty)
                        % cite.sty's \cite will automatically add leading
                        % space, if needed. Use cite.sty's noadjust option
                        % (cite.sty V3.8 and later) if you want to turn this
                        % off. cite.sty is already installed on most LaTeX
                        % systems. The latest version can be obtained at:
                        % http://www.ctan.org/tex-archive/macros/latex/contrib/supported/cite/

\usepackage{graphicx}   % Written by David Carlisle and Sebastian Rahtz
                        % Required if you want graphics, photos, etc.
                        % graphicx.sty is already installed on most LaTeX
                        % systems. The latest version and documentation can
                        % be obtained at:
                        % http://www.ctan.org/tex-archive/macros/latex/required/graphics/
                        % Another good source of documentation is "Using
                        % Imported Graphics in LaTeX2e" by Keith Reckdahl
                        % which can be found as esplatex.ps and epslatex.pdf
                        % at: http://www.ctan.org/tex-archive/info/

%\usepackage{psfrag}    % Written by Craig Barratt, Michael C. Grant,
                        % and David Carlisle
                        % This package allows you to substitute LaTeX
                        % commands for text in imported EPS graphic files.
                        % In this way, LaTeX symbols can be placed into
                        % graphics that have been generated by other
                        % applications. You must use latex->dvips->ps2pdf
                        % workflow (not direct pdf output from pdflatex) if
                        % you wish to use this capability because it works
                        % via some PostScript tricks. Alternatively, the
                        % graphics could be processed as separate files via
                        % psfrag and dvips, then converted to PDF for
                        % inclusion in the main file which uses pdflatex.
                        % Docs are in "The PSfrag System" by Michael C. Grant
                        % and David Carlisle. There is also some information
                        % about using psfrag in "Using Imported Graphics in
                        % LaTeX2e" by Keith Reckdahl which documents the
                        % graphicx package (see above). The psfrag package
                        % and documentation can be obtained at:
                        % http://www.ctan.org/tex-archive/macros/latex/contrib/supported/psfrag/

%\usepackage{subfigure} % Written by Steven Douglas Cochran
                        % This package makes it easy to put subfigures
                        % in your figures. i.e., "figure 1a and 1b"
                        % Docs are in "Using Imported Graphics in LaTeX2e"
                        % by Keith Reckdahl which also documents the graphicx
                        % package (see above). subfigure.sty is already
                        % installed on most LaTeX systems. The latest version
                        % and documentation can be obtained at:
                        % http://www.ctan.org/tex-archive/macros/latex/contrib/supported/subfigure/

\usepackage{url}        % Written by Donald Arseneau
                        % Provides better support for handling and breaking
                        % URLs. url.sty is already installed on most LaTeX
                        % systems. The latest version can be obtained at:
                        % http://www.ctan.org/tex-archive/macros/latex/contrib/other/misc/
                        % Read the url.sty source comments for usage information.

%\usepackage{stfloats}  % Written by Sigitas Tolusis
                        % Gives LaTeX2e the ability to do double column
                        % floats at the bottom of the page as well as the top.
                        % (e.g., "\begin{figure*}[!b]" is not normally
                        % possible in LaTeX2e). This is an invasive package
                        % which rewrites many portions of the LaTeX2e output
                        % routines. It may not work with other packages that
                        % modify the LaTeX2e output routine and/or with other
                        % versions of LaTeX. The latest version and
                        % documentation can be obtained at:
                        % http://www.ctan.org/tex-archive/macros/latex/contrib/supported/sttools/
                        % Documentation is contained in the stfloats.sty
                        % comments as well as in the presfull.pdf file.
                        % Do not use the stfloats baselinefloat ability as
                        % IEEE does not allow \baselineskip to stretch.
                        % Authors submitting work to the IEEE should note
                        % that IEEE rarely uses double column equations and
                        % that authors should try to avoid such use.
                        % Do not be tempted to use the cuted.sty or
                        % midfloat.sty package (by the same author) as IEEE
                        % does not format its papers in such ways.

\usepackage{amsmath}    % From the American Mathematical Society
                        % A popular package that provides many helpful commands
                        % for dealing with mathematics. Note that the AMSmath
                        % package sets \interdisplaylinepenalty to 10000 thus
                        % preventing page breaks from occurring within multiline
                        % equations. Use:
%\interdisplaylinepenalty=2500
                        % after loading amsmath to restore such page breaks
                        % as IEEEtran.cls normally does. amsmath.sty is already
                        % installed on most LaTeX systems. The latest version
                        % and documentation can be obtained at:
                        % http://www.ctan.org/tex-archive/macros/latex/required/amslatex/math/



% Other popular packages for formatting tables and equations include:

%\usepackage{array}
% Frank Mittelbach's and David Carlisle's array.sty which improves the
% LaTeX2e array and tabular environments to provide better appearances and
% additional user controls. array.sty is already installed on most systems.
% The latest version and documentation can be obtained at:
% http://www.ctan.org/tex-archive/macros/latex/required/tools/

% V1.6 of IEEEtran contains the IEEEeqnarray family of commands that can
% be used to generate multiline equations as well as matrices, tables, etc.

% Also of notable interest:
% Scott Pakin's eqparbox package for creating (automatically sized) equal
% width boxes. Available:
% http://www.ctan.org/tex-archive/macros/latex/contrib/supported/eqparbox/

% *** Do not adjust lengths that control margins, column widths, etc. ***
% *** Do not use packages that alter fonts (such as pslatex).         ***
% There should be no need to do such things with IEEEtran.cls V1.6 and later.

\usepackage{booktabs}
\usepackage{xcolor}
\usepackage{array}
\usepackage{caption}
\usepackage{arydshln}
\usepackage{amsfonts}

\def\one{\mathbf{1}}

\setlength{\dashlinedash}{.4pt}
\setlength{\dashlinegap}{.8pt}

\newcommand{\ct}[1]{\multicolumn{1}{c}{#1}}

\def\vsep{\vspace{3mm}\\}

\newcommand{\rpm}{\sbox0{$1$}\sbox2{$\scriptstyle\pm$}
	\raise\dimexpr(\ht0-\ht2)/2\relax\box2 } % plus ou moins
\newcommand{\h}[1]{\colorbox{yellow}{#1}}

% Your document starts here!
\usepackage{Sweave}
\begin{document}

% Define document title and author
\title{\Large{Identifying Relevant Genes Related Atopic Dermatitis  to  using Transcriptomic}}
\author{%
\begin{tabular}{c} Creel, Kristin \\ ISPED \\ University of Bordeaux \\ \texttt{kristin.creel@etu.u-bordeaux.fr} \end{tabular} \and
\begin{tabular}{c} Dong, Larry \\ Department of Epidemiology, Biostatistics \\ and Occupational Health \\ McGill University \\ \texttt{larry.dong@mail.mcgill.ca} \end{tabular} \and
\begin{tabular}{c} Nama Ravi, Sneha Keerthi \\ ISPED \\ University of Bordeaux \\ \texttt{sneha-keerthi.nama-ravi@etu.u-bordeaux.fr} \end{tabular}}
\maketitle

% Write abstract here


\input{version-1-concordance}

\section*{Abstract}

\textbf{Background --- }

\textbf{Methods ---} 

\textbf{Results ---} 

\textbf{Conclusion ---}

\textbf{Keywords:} Atopic dermatitis, transcriptomics, differential gene expression, unsupervised learning, prediction model.

\section{Introduction}

Atopic dermatitis (AD) or atopic eczema is an itchy, inflammatory skin condition characterised by poorly defined erythema with edema vesicles, and weeping in the acute stage and lichenification in the chronic stage. The global prevalence is 15-20 $\%$ in children and 1-3$\%$ in adults, posing a significant burden on health-care resources and patientsâ€™ quality of life \cite{nutten2015atopic}.\\

The etiological factors associated with the initiation and progress of the disease are known to be genetic, environmental and immunological that affects the epithelial barrier-immunity interplay \cite{peng2015pathogenesis}.\\

Clinical investigations and discoveries in molecular medicine have positively identified 46 genes linked to AD. Mutations in filaggrin (FLG) genes (influencing intermediate filament protein filaggrin expression) are most common in the AD diseased population, it affects 10-50$\%$ of AD patients worldwide.\\

Few additional barrier genes encoded by the epidermal differentiation complex (EDC) locus chromosome 1q21, including claudins, loricrin (LOR), involucrin (IVL), SPINK5, AND tmem79/matt, are also associated with AD.\\

The genes of innate immune system like NOD1, NOD2, TLR2, CD14, and DEFB1, that encode the integral factors in cutaneous immunologic response to non- specific antigens may also experience mutations and cause AD \cite{guttman2017atopic}.\\

Studies have identified FLG gene mutation to be the most significant risk factor for AD, followed by the genes in the type 2 T helper lymphocyte (Th2) signalling pathways.\\

Additionally, gene profiling assays demonstrated AD is associated with decreased gene expression of epidermal differentiation complex genes and elevated Th2 and Th17 genes. Hypomethylation of TSLP and FCER1G in AD were also reported; and miR-155, which targets the immune suppressor CTLA-4, was found to be significantly over-expressed in infiltrating T cells in AD skin lesions \cite{guttman2017atopic, bin2016genetic}.\\

This study attempts to examine the degree of dissimilarity between the transcriptomic expression between lesional and non-lesional samples in participants with atopic dermatitis. (ELABORATE MORE HERE...)

\section{Materials and Methods}

\subsection{Data Source}

The data at hand comes from the Microbes in Allergy and Autoimmunity Related to the Skin (MAARS). The purpose of this multidisciplinary research consortium is to better understand the characteristics surrounding two major chronic inflammatory skin diseases: atopic dermatitis and psoriasis\cite{MAARS}. In the MAARS study, these two illnesses that (AFFECT X PEOPLE IN THE WORLD) are investigated from different scientific perspectives and one of the research directions involved exploring the transcriptomic data of lesional and non-lesional samples in patients with atopic dermatitis, psoriasis and neither of the two studied skin diseases.

% the relationship between the host and its microbiome 

\subsubsection*{Data Management}

Demographic data for all patients were available alongside their known allergies, concomitant medication and transcriptomic data for sequenced samples. Because it is only of interest to assess RNA expression levels in samples from patients with atopic dermatitis, two sets of data was discarded prior to conducting subsequent analyses: participants with AD who have no sequenced samples and genetic information for samples from healthy individuals and ones with psoriasis. The normalization procedure for transcriptomic data was already performed prior to obtaining the data for this anlysis. No genetic information was missing, so statistical methods for imputation was not needed for this study.


\subsection{Exploratory Data Analysis}

Exploratory data analysis was conducted to better understand the data at hand from a data-driven perspective. To this end, an MA diagram was plotted alongside several clustering methods which were used to help depict possible underlying patterns from the MAARS transcriptomic data: principal component analysis (PCA) and hierarchical clustering\cite{maaten2008visualizing}\cite{friedman2001elements}. In statistical machine learning, clustering techniques falls under unsupervised learning, which is a paradigm that attempts to perform inference on data without response variables or ``labels" \cite{friedman2001elements}. In the recent years, this area of machine learning has gathered much attention from researchers; data separability without the use of labels is known as disentangling in the autoencoder literature\cite{higgins2017beta}\cite{burgess2018understanding}. Here, the sample labels are the lesional status of the skin sample and the primary purpose of performing such unsupervised analyses was to visually assess the separability of the samples using their transcriptomic information.

\subsection{Statistical Analysis}

\subsubsection{Differential Gene Expression Analysis}

The overarching goal of the statistical analysis was the determine the relevant genes from a transcriptomic datatset that are predictive of AD. The analysis plan was performed two-fold; the identification of genes highly associated with AD was first performed using differential gene expression analysis within a multiple hypothesis testing framework. The transcriptomic data was available as a continuous variable since the microarray data has been normalized.\\

Let $Y_{ij} \in \mathbb{R}$ denote the gene expression level of gene $j$ in sample $i$ and let $X_i \in \{0, 1\}$ represent its lesional status. The following linear model was fitted:
\begin{align*}
  Y_{ij} &= \beta_{0j} + \beta_{1j}X_i + \sum_{k:\,k\in\mathcal{P}}\beta_{k}\one_{pat(i) = k} + \epsilon_{ij}\\
  &=\beta_{0j} + \beta_{1j}X_i + \beta_{pat(i)}\one_{pat(i)} + \epsilon_{ij}
\end{align*}

where the $\mathcal{P}$ is the set of all patients and $pat(i) \in \mathcal{P}$ is used to represent the patient to which belongs sample $i$. The coefficient $\beta_k$ accounts for the intersample correlation for samples which are obtained from the same patient. By construction, $\one_{pat(i) = k} = 1$ for exactly one $k \in \mathcal{P}$ and $\big\vert \mathcal{P}\big\vert < n$.\\

Originally developped for two-color microarray experiments, it is now commonplace to employ an empirical Bayes method to analyze differential gene expression with linear models like the one above\cite{smyth2004linear}. In a hypothesis framework, the following distributional assumptions for $\beta_{1j}$ are assumed for all genes $j$:
\begin{align*}
  H_0:\qquad \beta_{1j} \,\big\vert\, \sigma^2_j &\sim \mathcal{N}\left(0, \sigma^2_j\right)\\
  H_1:\qquad \beta_{1j} \,\big\vert\, \sigma^2_j &\,\,\cancel{\sim}\,\, \mathcal{N}\left(0, \sigma^2_j\right)
\end{align*}

whereby $p$-values were ultimately obtained and adjusted using the Benjamini-Hochberg procedure to reduce the error in false discovery rate prone to be inflated due to performing multiple parallel statistical tests\cite{benjamini2010discovering}. A significance level of $0.05$ was used in determining the genes highly associated with the outcome of interest, the lesional status of sequenced samples; a volcano plot was used to highlight potential significant genes.\\

A ensemble learning prediction model was then implemented to assess predictability of sample lesional status; this binary model was chosen due to its strengths in obtaining better predictions as it weights multiple models according to how well each perform on this specific task\cite{friedman2001elements}. MENTION VARIABLE SELECTION. The receiving operating characteristic (ROC) curve is often used to evaluate data-driven binary prediction models and the area under this curve (AUROC) was used here to quantify the goodness of the prediction model\cite{hanley1982meaning}. Internal validation was performed using leave-one-out cross-validation method whereby the AUROC was evaluated at every iteration\cite{efron1994introduction}. %An additional sensitivity analysis of the prediction model was conducted via bootstrap resampling whereby a thousand samples of the data were sampled without replacement to create an empirical $95\%$ confidence interval for the overall AUROC\cite{efron1994introduction}.

%This goodness of fit metric was used on ensemble learning model \cite{efron1994introduction}\\

\section{Results}

In Fig. 1, a simple 2D visualization of the results obtained from PCA is available, whereby lesional samples are displayed in black and non-lesional samples are displayed in red.

\begin{figure}[!htp]
    \begin{center}
    \begin{minipage}{0.5 \textwidth}
      \centering
      \includegraphics[width=\textwidth]{../exploratory-data-analysis/pca-plot.png}
      \label{pca}
    \end{minipage}
  \end{center}
\end{figure}


The horizontal axis represents the first component obtained through PCA and the vertical axis the second; they respectively explain $8.78\%$ and $6.02\%$ of the variance in the transcriptomic data.

\begin{figure}[!htp]
  \begin{center}
    \begin{minipage}{0.5 \textwidth}
      \centering
      \includegraphics[width=\textwidth]{../analysis/Volcqno-plot1.png}
      \label{volcano}
    \end{minipage}
  \end{center}
\end{figure}

\begin{figure}[!htp]
  \begin{center}
    \begin{minipage}{0.5 \textwidth}
      \centering
      \includegraphics[width=\textwidth]{../analysis/MA-plot-AD-paired-design.png}
      \label{MA}
    \end{minipage}
  \end{center}
\end{figure}

\section{Discussion}

\section{Conclusion}


\bibliographystyle{plain}
\bibliography{References}

\end{document}
