{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## `exploratory-data-analysis.ipynb`\n",
    "\n",
    "The purpose of this Jupyter notebook is to simply explore and have a look at the dataset."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "%matplotlib inline\n",
    "%config InlineBackend.figure_format = 'retina'\n",
    "plt.style.use('seaborn-darkgrid')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>clinical_group</th>\n",
       "      <th>anatomical_location</th>\n",
       "      <th>anatomical_location_label</th>\n",
       "      <th>lesional</th>\n",
       "      <th>MAARS_identifier</th>\n",
       "      <th>Institution</th>\n",
       "      <th>CUSTOM_Age</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Known_Allergies_v2..Pseudo_Drug_Allergy</th>\n",
       "      <th>...</th>\n",
       "      <th>Other_concurrent_chronic_diseases_v2..Others</th>\n",
       "      <th>Other_concurrent_chronic_diseases_v2..Diabetes_.non.insulin.</th>\n",
       "      <th>Other_concurrent_chronic_diseases_v2..Thyroid_dysfunction</th>\n",
       "      <th>Other_concurrent_chronic_diseases_v2..Asthma</th>\n",
       "      <th>Other_concurrent_chronic_diseases_v2..Hypertension</th>\n",
       "      <th>Global_Assessment_Score</th>\n",
       "      <th>CUSTOM_Malignancies_._skin</th>\n",
       "      <th>CUSTOM_Malignancies_._other</th>\n",
       "      <th>CUSTOM_Fam._hist._Atopic_dermatitis</th>\n",
       "      <th>CUSTOM_Family_History_of_Psoriasis</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>MAARS_3_018_02</td>\n",
       "      <td>AD</td>\n",
       "      <td>posterior_thigh</td>\n",
       "      <td>PT</td>\n",
       "      <td>NON_LES</td>\n",
       "      <td>MAARS_3_018</td>\n",
       "      <td>UH</td>\n",
       "      <td>25</td>\n",
       "      <td>Male</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>moderate</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>MAARS_3_018_01</td>\n",
       "      <td>AD</td>\n",
       "      <td>posterior_thigh</td>\n",
       "      <td>PT</td>\n",
       "      <td>LES</td>\n",
       "      <td>MAARS_3_018</td>\n",
       "      <td>UH</td>\n",
       "      <td>25</td>\n",
       "      <td>Male</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>moderate</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>MAARS_3_018_03</td>\n",
       "      <td>AD</td>\n",
       "      <td>posterior_thigh</td>\n",
       "      <td>PT</td>\n",
       "      <td>LES</td>\n",
       "      <td>MAARS_3_018</td>\n",
       "      <td>UH</td>\n",
       "      <td>25</td>\n",
       "      <td>Male</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>moderate</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>MAARS_3_018_04</td>\n",
       "      <td>AD</td>\n",
       "      <td>posterior_thigh</td>\n",
       "      <td>PT</td>\n",
       "      <td>NON_LES</td>\n",
       "      <td>MAARS_3_018</td>\n",
       "      <td>UH</td>\n",
       "      <td>25</td>\n",
       "      <td>Male</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>moderate</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>MAARS_3_082_02</td>\n",
       "      <td>AD</td>\n",
       "      <td>posterior_thigh</td>\n",
       "      <td>PT</td>\n",
       "      <td>NON_LES</td>\n",
       "      <td>MAARS_3_082</td>\n",
       "      <td>UH</td>\n",
       "      <td>62</td>\n",
       "      <td>Male</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>moderate</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        sample_id clinical_group anatomical_location  \\\n",
       "0  MAARS_3_018_02             AD     posterior_thigh   \n",
       "1  MAARS_3_018_01             AD     posterior_thigh   \n",
       "2  MAARS_3_018_03             AD     posterior_thigh   \n",
       "3  MAARS_3_018_04             AD     posterior_thigh   \n",
       "4  MAARS_3_082_02             AD     posterior_thigh   \n",
       "\n",
       "  anatomical_location_label lesional MAARS_identifier Institution  CUSTOM_Age  \\\n",
       "0                        PT  NON_LES      MAARS_3_018          UH          25   \n",
       "1                        PT      LES      MAARS_3_018          UH          25   \n",
       "2                        PT      LES      MAARS_3_018          UH          25   \n",
       "3                        PT  NON_LES      MAARS_3_018          UH          25   \n",
       "4                        PT  NON_LES      MAARS_3_082          UH          62   \n",
       "\n",
       "  Gender  Known_Allergies_v2..Pseudo_Drug_Allergy  ...  \\\n",
       "0   Male                                    False  ...   \n",
       "1   Male                                    False  ...   \n",
       "2   Male                                    False  ...   \n",
       "3   Male                                    False  ...   \n",
       "4   Male                                    False  ...   \n",
       "\n",
       "   Other_concurrent_chronic_diseases_v2..Others  \\\n",
       "0                                         False   \n",
       "1                                         False   \n",
       "2                                         False   \n",
       "3                                         False   \n",
       "4                                         False   \n",
       "\n",
       "   Other_concurrent_chronic_diseases_v2..Diabetes_.non.insulin.  \\\n",
       "0                                              False              \n",
       "1                                              False              \n",
       "2                                              False              \n",
       "3                                              False              \n",
       "4                                              False              \n",
       "\n",
       "   Other_concurrent_chronic_diseases_v2..Thyroid_dysfunction  \\\n",
       "0                                              False           \n",
       "1                                              False           \n",
       "2                                              False           \n",
       "3                                              False           \n",
       "4                                              False           \n",
       "\n",
       "   Other_concurrent_chronic_diseases_v2..Asthma  \\\n",
       "0                                         False   \n",
       "1                                         False   \n",
       "2                                         False   \n",
       "3                                         False   \n",
       "4                                         False   \n",
       "\n",
       "   Other_concurrent_chronic_diseases_v2..Hypertension  \\\n",
       "0                                              False    \n",
       "1                                              False    \n",
       "2                                              False    \n",
       "3                                              False    \n",
       "4                                              False    \n",
       "\n",
       "   Global_Assessment_Score  CUSTOM_Malignancies_._skin  \\\n",
       "0                 moderate                       False   \n",
       "1                 moderate                       False   \n",
       "2                 moderate                       False   \n",
       "3                 moderate                       False   \n",
       "4                 moderate                       False   \n",
       "\n",
       "   CUSTOM_Malignancies_._other  CUSTOM_Fam._hist._Atopic_dermatitis  \\\n",
       "0                        False                                False   \n",
       "1                        False                                False   \n",
       "2                        False                                False   \n",
       "3                        False                                False   \n",
       "4                        False                                 True   \n",
       "\n",
       "   CUSTOM_Family_History_of_Psoriasis  \n",
       "0                                 NaN  \n",
       "1                                 NaN  \n",
       "2                                 NaN  \n",
       "3                                 NaN  \n",
       "4                                 NaN  \n",
       "\n",
       "[5 rows x 34 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data1 = pd.read_csv('../data/M2PHDS_19-20_OMICS_CLIN_DATA_MAARS_all_Fri_Apr_04_14h_CEST_2014.csv', sep='\\t')\n",
    "data1.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient#Identification#MAARS identifier (MAARS_identifier)</th>\n",
       "      <th>patient#Identification#Date of first visit (Date_of_first_visit)</th>\n",
       "      <th>patient#Identification#Institution (Institution)</th>\n",
       "      <th>patient#Identification#Physician (Physician)</th>\n",
       "      <th>patient#Inclusion/Exclusion criteria#Inclusion criteria#Written informed consent (Written_informed_consent)</th>\n",
       "      <th>patient#Inclusion/Exclusion criteria#Inclusion criteria#18 years old or more (18_years_old_or_more)</th>\n",
       "      <th>patient#Inclusion/Exclusion criteria#Inclusion criteria#Patients with Atopic Dermatitis, diagnosed using Hanifin-Rajka Criteria (Patients_with_Atopic_Dermatitis,_diagnosed_using_Hanifin-Rajka_Criteria)</th>\n",
       "      <th>patient#Inclusion/Exclusion criteria#Exclusion criteria#Subject unable to give written informed consent (Subject_unable_to_give_written_informed_consent)</th>\n",
       "      <th>patient#Inclusion/Exclusion criteria#Exclusion criteria#Subject has no allergen-specific IgE and no allergen-specific immediate type reactions (Subject_has_no_allergen-specific_IgE_and_no_allergen-specific_immediate_type_reactions)</th>\n",
       "      <th>patient#Inclusion/Exclusion criteria#Exclusion criteria#Subjects who have received treatment at the biopsy site at least 2 weeks prior to screening (Subjects_who_have_received_treatment_at_the_biopsy_site_at_least_2_weeks_prior_to_screening)</th>\n",
       "      <th>...</th>\n",
       "      <th>blood for genomic DNA#blood for genomic DNA#Excluded (Excluded)</th>\n",
       "      <th>blood for genomic DNA#blood for genomic DNA#Exclusion comments (Exclusion_comments)</th>\n",
       "      <th>blood for PBMC#blood for PBMC#MAARS Sample identifier (MAARS_Sample_identifier)</th>\n",
       "      <th>blood for PBMC#blood for PBMC#Sampling date (Sampling_date)</th>\n",
       "      <th>blood for PBMC#blood for PBMC#Excluded (Excluded)</th>\n",
       "      <th>blood for PBMC#blood for PBMC#Exclusion comments (Exclusion_comments)</th>\n",
       "      <th>plasma#plasma#MAARS Sample identifier (MAARS_Sample_identifier)</th>\n",
       "      <th>plasma#plasma#Sampling date (Sampling_date)</th>\n",
       "      <th>plasma#plasma#Excluded (Excluded)</th>\n",
       "      <th>plasma#plasma#Exclusion comments (Exclusion_comments)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>MAARS_1_003</td>\n",
       "      <td>06/12/12</td>\n",
       "      <td>HHU</td>\n",
       "      <td>Homey</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MAARS_1_003_08</td>\n",
       "      <td>06/12/12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MAARS_1_003_09</td>\n",
       "      <td>06/12/12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>MAARS_1_008</td>\n",
       "      <td>15/02/12</td>\n",
       "      <td>HHU</td>\n",
       "      <td>Baran</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MAARS_1_008_08</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MAARS_1_008_09</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>MAARS_1_016</td>\n",
       "      <td>14/03/12</td>\n",
       "      <td>HHU</td>\n",
       "      <td>Baran</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MAARS_1_016_08</td>\n",
       "      <td>09/12/11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MAARS_1_016_09</td>\n",
       "      <td>09/12/11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>MAARS_1_017</td>\n",
       "      <td>14/03/12</td>\n",
       "      <td>HHU</td>\n",
       "      <td>Baran</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MAARS_1_017_08</td>\n",
       "      <td>14/03/12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MAARS_1_017_09</td>\n",
       "      <td>14/03/12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>MAARS_1_044</td>\n",
       "      <td>09/05/12</td>\n",
       "      <td>HHU</td>\n",
       "      <td>Jahn</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MAARS_1_044_08</td>\n",
       "      <td>09/05/12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MAARS_1_044_09</td>\n",
       "      <td>09/05/12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 297 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  patient#Identification#MAARS identifier (MAARS_identifier)  \\\n",
       "0                                        MAARS_1_003           \n",
       "1                                        MAARS_1_008           \n",
       "2                                        MAARS_1_016           \n",
       "3                                        MAARS_1_017           \n",
       "4                                        MAARS_1_044           \n",
       "\n",
       "  patient#Identification#Date of first visit (Date_of_first_visit)  \\\n",
       "0                                           06/12/12                 \n",
       "1                                           15/02/12                 \n",
       "2                                           14/03/12                 \n",
       "3                                           14/03/12                 \n",
       "4                                           09/05/12                 \n",
       "\n",
       "  patient#Identification#Institution (Institution)  \\\n",
       "0                                              HHU   \n",
       "1                                              HHU   \n",
       "2                                              HHU   \n",
       "3                                              HHU   \n",
       "4                                              HHU   \n",
       "\n",
       "  patient#Identification#Physician (Physician)  \\\n",
       "0                                        Homey   \n",
       "1                                        Baran   \n",
       "2                                        Baran   \n",
       "3                                        Baran   \n",
       "4                                         Jahn   \n",
       "\n",
       "  patient#Inclusion/Exclusion criteria#Inclusion criteria#Written informed consent (Written_informed_consent)  \\\n",
       "0                                                Yes                                                            \n",
       "1                                                Yes                                                            \n",
       "2                                                Yes                                                            \n",
       "3                                                Yes                                                            \n",
       "4                                                Yes                                                            \n",
       "\n",
       "  patient#Inclusion/Exclusion criteria#Inclusion criteria#18 years old or more (18_years_old_or_more)  \\\n",
       "0                                                Yes                                                    \n",
       "1                                                Yes                                                    \n",
       "2                                                Yes                                                    \n",
       "3                                                Yes                                                    \n",
       "4                                                Yes                                                    \n",
       "\n",
       "  patient#Inclusion/Exclusion criteria#Inclusion criteria#Patients with Atopic Dermatitis, diagnosed using Hanifin-Rajka Criteria (Patients_with_Atopic_Dermatitis,_diagnosed_using_Hanifin-Rajka_Criteria)  \\\n",
       "0                                                Yes                                                                                                                                                          \n",
       "1                                                Yes                                                                                                                                                          \n",
       "2                                                Yes                                                                                                                                                          \n",
       "3                                                Yes                                                                                                                                                          \n",
       "4                                                Yes                                                                                                                                                          \n",
       "\n",
       "  patient#Inclusion/Exclusion criteria#Exclusion criteria#Subject unable to give written informed consent (Subject_unable_to_give_written_informed_consent)  \\\n",
       "0                                                 No                                                                                                          \n",
       "1                                                 No                                                                                                          \n",
       "2                                                 No                                                                                                          \n",
       "3                                                 No                                                                                                          \n",
       "4                                                 No                                                                                                          \n",
       "\n",
       "  patient#Inclusion/Exclusion criteria#Exclusion criteria#Subject has no allergen-specific IgE and no allergen-specific immediate type reactions (Subject_has_no_allergen-specific_IgE_and_no_allergen-specific_immediate_type_reactions)  \\\n",
       "0                                                 No                                                                                                                                                                                        \n",
       "1                                                 No                                                                                                                                                                                        \n",
       "2                                                 No                                                                                                                                                                                        \n",
       "3                                                 No                                                                                                                                                                                        \n",
       "4                                                 No                                                                                                                                                                                        \n",
       "\n",
       "  patient#Inclusion/Exclusion criteria#Exclusion criteria#Subjects who have received treatment at the biopsy site at least 2 weeks prior to screening (Subjects_who_have_received_treatment_at_the_biopsy_site_at_least_2_weeks_prior_to_screening)  \\\n",
       "0                                                 No                                                                                                                                                                                                  \n",
       "1                                                 No                                                                                                                                                                                                  \n",
       "2                                                 No                                                                                                                                                                                                  \n",
       "3                                                 No                                                                                                                                                                                                  \n",
       "4                                                 No                                                                                                                                                                                                  \n",
       "\n",
       "   ... blood for genomic DNA#blood for genomic DNA#Excluded (Excluded)  \\\n",
       "0  ...                                                NaN                \n",
       "1  ...                                                NaN                \n",
       "2  ...                                                NaN                \n",
       "3  ...                                                NaN                \n",
       "4  ...                                                NaN                \n",
       "\n",
       "  blood for genomic DNA#blood for genomic DNA#Exclusion comments (Exclusion_comments)  \\\n",
       "0                                                NaN                                    \n",
       "1                                                NaN                                    \n",
       "2                                                NaN                                    \n",
       "3                                                NaN                                    \n",
       "4                                                NaN                                    \n",
       "\n",
       "  blood for PBMC#blood for PBMC#MAARS Sample identifier (MAARS_Sample_identifier)  \\\n",
       "0                                     MAARS_1_003_08                                \n",
       "1                                     MAARS_1_008_08                                \n",
       "2                                     MAARS_1_016_08                                \n",
       "3                                     MAARS_1_017_08                                \n",
       "4                                     MAARS_1_044_08                                \n",
       "\n",
       "   blood for PBMC#blood for PBMC#Sampling date (Sampling_date)  \\\n",
       "0                                           06/12/12             \n",
       "1                                                NaN             \n",
       "2                                           09/12/11             \n",
       "3                                           14/03/12             \n",
       "4                                           09/05/12             \n",
       "\n",
       "  blood for PBMC#blood for PBMC#Excluded (Excluded)  \\\n",
       "0                                               NaN   \n",
       "1                                               NaN   \n",
       "2                                               NaN   \n",
       "3                                               NaN   \n",
       "4                                               NaN   \n",
       "\n",
       "  blood for PBMC#blood for PBMC#Exclusion comments (Exclusion_comments)  \\\n",
       "0                                                NaN                      \n",
       "1                                                NaN                      \n",
       "2                                                NaN                      \n",
       "3                                                NaN                      \n",
       "4                                                NaN                      \n",
       "\n",
       "  plasma#plasma#MAARS Sample identifier (MAARS_Sample_identifier)  \\\n",
       "0                                     MAARS_1_003_09                \n",
       "1                                     MAARS_1_008_09                \n",
       "2                                     MAARS_1_016_09                \n",
       "3                                     MAARS_1_017_09                \n",
       "4                                     MAARS_1_044_09                \n",
       "\n",
       "  plasma#plasma#Sampling date (Sampling_date)  \\\n",
       "0                                    06/12/12   \n",
       "1                                         NaN   \n",
       "2                                    09/12/11   \n",
       "3                                    14/03/12   \n",
       "4                                    09/05/12   \n",
       "\n",
       "  plasma#plasma#Excluded (Excluded)  \\\n",
       "0                               NaN   \n",
       "1                               NaN   \n",
       "2                               NaN   \n",
       "3                               NaN   \n",
       "4                               NaN   \n",
       "\n",
       "  plasma#plasma#Exclusion comments (Exclusion_comments)  \n",
       "0                                                NaN     \n",
       "1                                                NaN     \n",
       "2                                                NaN     \n",
       "3                                                NaN     \n",
       "4                                                NaN     \n",
       "\n",
       "[5 rows x 297 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data2 = pd.read_csv('../data/M2PHDS_19-20_OMICS_CLIN_DATA_MAARS_AD_full_20190131_12-34-49.csv', sep='\\t')\n",
    "data2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MAARS_3_033_03</th>\n",
       "      <th>MAARS_3_032_03</th>\n",
       "      <th>MAARS_3_032_04</th>\n",
       "      <th>MAARS_2_012_03</th>\n",
       "      <th>MAARS_3_007_03</th>\n",
       "      <th>MAARS_3_007_04</th>\n",
       "      <th>MAARS_2_012_04</th>\n",
       "      <th>MAARS_2_004_03</th>\n",
       "      <th>MAARS_2_004_04</th>\n",
       "      <th>MAARS_2_007_04</th>\n",
       "      <th>...</th>\n",
       "      <th>MAARS_1_128_03</th>\n",
       "      <th>MAARS_2_083_03</th>\n",
       "      <th>MAARS_1_010_04</th>\n",
       "      <th>MAARS_1_036_04</th>\n",
       "      <th>MAARS_3_099_04</th>\n",
       "      <th>MAARS_1_076_03</th>\n",
       "      <th>MAARS_1_065_03</th>\n",
       "      <th>MAARS_2_044_03</th>\n",
       "      <th>MAARS_2_056_03</th>\n",
       "      <th>MAARS_1_098_04</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>ENSG00000000003_at</td>\n",
       "      <td>7.967</td>\n",
       "      <td>7.634</td>\n",
       "      <td>7.893</td>\n",
       "      <td>7.106</td>\n",
       "      <td>6.807</td>\n",
       "      <td>7.375</td>\n",
       "      <td>8.027</td>\n",
       "      <td>7.948</td>\n",
       "      <td>7.606</td>\n",
       "      <td>8.432</td>\n",
       "      <td>...</td>\n",
       "      <td>8.813</td>\n",
       "      <td>7.868</td>\n",
       "      <td>8.643</td>\n",
       "      <td>8.549</td>\n",
       "      <td>8.358</td>\n",
       "      <td>8.157</td>\n",
       "      <td>7.828</td>\n",
       "      <td>7.866</td>\n",
       "      <td>7.705</td>\n",
       "      <td>7.837</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSG00000000005_at</td>\n",
       "      <td>5.075</td>\n",
       "      <td>4.886</td>\n",
       "      <td>5.312</td>\n",
       "      <td>4.187</td>\n",
       "      <td>3.958</td>\n",
       "      <td>3.983</td>\n",
       "      <td>6.050</td>\n",
       "      <td>3.980</td>\n",
       "      <td>4.163</td>\n",
       "      <td>6.236</td>\n",
       "      <td>...</td>\n",
       "      <td>4.981</td>\n",
       "      <td>4.004</td>\n",
       "      <td>4.451</td>\n",
       "      <td>4.586</td>\n",
       "      <td>6.048</td>\n",
       "      <td>5.950</td>\n",
       "      <td>5.962</td>\n",
       "      <td>5.474</td>\n",
       "      <td>4.281</td>\n",
       "      <td>4.097</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSG00000000419_at</td>\n",
       "      <td>9.043</td>\n",
       "      <td>9.411</td>\n",
       "      <td>9.624</td>\n",
       "      <td>9.430</td>\n",
       "      <td>9.809</td>\n",
       "      <td>9.696</td>\n",
       "      <td>9.809</td>\n",
       "      <td>9.859</td>\n",
       "      <td>9.582</td>\n",
       "      <td>9.477</td>\n",
       "      <td>...</td>\n",
       "      <td>9.726</td>\n",
       "      <td>9.764</td>\n",
       "      <td>9.668</td>\n",
       "      <td>9.274</td>\n",
       "      <td>9.669</td>\n",
       "      <td>9.471</td>\n",
       "      <td>9.382</td>\n",
       "      <td>9.816</td>\n",
       "      <td>9.781</td>\n",
       "      <td>10.130</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSG00000000457_at</td>\n",
       "      <td>7.154</td>\n",
       "      <td>7.111</td>\n",
       "      <td>7.161</td>\n",
       "      <td>7.247</td>\n",
       "      <td>6.846</td>\n",
       "      <td>7.076</td>\n",
       "      <td>7.254</td>\n",
       "      <td>6.941</td>\n",
       "      <td>7.127</td>\n",
       "      <td>6.722</td>\n",
       "      <td>...</td>\n",
       "      <td>7.214</td>\n",
       "      <td>7.032</td>\n",
       "      <td>7.088</td>\n",
       "      <td>7.218</td>\n",
       "      <td>7.187</td>\n",
       "      <td>7.007</td>\n",
       "      <td>7.116</td>\n",
       "      <td>7.023</td>\n",
       "      <td>6.718</td>\n",
       "      <td>7.182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ENSG00000000460_at</td>\n",
       "      <td>5.992</td>\n",
       "      <td>5.819</td>\n",
       "      <td>5.931</td>\n",
       "      <td>5.881</td>\n",
       "      <td>5.975</td>\n",
       "      <td>5.288</td>\n",
       "      <td>5.236</td>\n",
       "      <td>5.928</td>\n",
       "      <td>5.781</td>\n",
       "      <td>5.601</td>\n",
       "      <td>...</td>\n",
       "      <td>5.497</td>\n",
       "      <td>5.511</td>\n",
       "      <td>5.524</td>\n",
       "      <td>5.773</td>\n",
       "      <td>5.419</td>\n",
       "      <td>5.756</td>\n",
       "      <td>5.967</td>\n",
       "      <td>5.363</td>\n",
       "      <td>5.513</td>\n",
       "      <td>6.068</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 618 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    MAARS_3_033_03  MAARS_3_032_03  MAARS_3_032_04  \\\n",
       "ENSG00000000003_at           7.967           7.634           7.893   \n",
       "ENSG00000000005_at           5.075           4.886           5.312   \n",
       "ENSG00000000419_at           9.043           9.411           9.624   \n",
       "ENSG00000000457_at           7.154           7.111           7.161   \n",
       "ENSG00000000460_at           5.992           5.819           5.931   \n",
       "\n",
       "                    MAARS_2_012_03  MAARS_3_007_03  MAARS_3_007_04  \\\n",
       "ENSG00000000003_at           7.106           6.807           7.375   \n",
       "ENSG00000000005_at           4.187           3.958           3.983   \n",
       "ENSG00000000419_at           9.430           9.809           9.696   \n",
       "ENSG00000000457_at           7.247           6.846           7.076   \n",
       "ENSG00000000460_at           5.881           5.975           5.288   \n",
       "\n",
       "                    MAARS_2_012_04  MAARS_2_004_03  MAARS_2_004_04  \\\n",
       "ENSG00000000003_at           8.027           7.948           7.606   \n",
       "ENSG00000000005_at           6.050           3.980           4.163   \n",
       "ENSG00000000419_at           9.809           9.859           9.582   \n",
       "ENSG00000000457_at           7.254           6.941           7.127   \n",
       "ENSG00000000460_at           5.236           5.928           5.781   \n",
       "\n",
       "                    MAARS_2_007_04  ...  MAARS_1_128_03  MAARS_2_083_03  \\\n",
       "ENSG00000000003_at           8.432  ...           8.813           7.868   \n",
       "ENSG00000000005_at           6.236  ...           4.981           4.004   \n",
       "ENSG00000000419_at           9.477  ...           9.726           9.764   \n",
       "ENSG00000000457_at           6.722  ...           7.214           7.032   \n",
       "ENSG00000000460_at           5.601  ...           5.497           5.511   \n",
       "\n",
       "                    MAARS_1_010_04  MAARS_1_036_04  MAARS_3_099_04  \\\n",
       "ENSG00000000003_at           8.643           8.549           8.358   \n",
       "ENSG00000000005_at           4.451           4.586           6.048   \n",
       "ENSG00000000419_at           9.668           9.274           9.669   \n",
       "ENSG00000000457_at           7.088           7.218           7.187   \n",
       "ENSG00000000460_at           5.524           5.773           5.419   \n",
       "\n",
       "                    MAARS_1_076_03  MAARS_1_065_03  MAARS_2_044_03  \\\n",
       "ENSG00000000003_at           8.157           7.828           7.866   \n",
       "ENSG00000000005_at           5.950           5.962           5.474   \n",
       "ENSG00000000419_at           9.471           9.382           9.816   \n",
       "ENSG00000000457_at           7.007           7.116           7.023   \n",
       "ENSG00000000460_at           5.756           5.967           5.363   \n",
       "\n",
       "                    MAARS_2_056_03  MAARS_1_098_04  \n",
       "ENSG00000000003_at           7.705           7.837  \n",
       "ENSG00000000005_at           4.281           4.097  \n",
       "ENSG00000000419_at           9.781          10.130  \n",
       "ENSG00000000457_at           6.718           7.182  \n",
       "ENSG00000000460_at           5.513           6.068  \n",
       "\n",
       "[5 rows x 618 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "norm_data = pd.read_csv('../data/M2PHDS_19-20_OMICS_TRANSC_MAARS_normTranscriptome_618samples_16042014.txt', sep='\\t')\n",
    "norm_data.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### All samples in gene dataset are available in `M2PHDS_19-20_OMICS_CLIN_DATA_MAARS_all_Fri_Apr_04_14h_CEST_2014.csv`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "618"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(data1['sample_id'].values.astype(str)).intersection(set(np.array(list(norm_data)).astype(str))))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "618"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sum([s in data1['sample_id'].values.astype(str) for s in np.array(list(norm_data)).astype(str)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1317, 34)\n",
      "(94, 297)\n",
      "(32633, 618)\n"
     ]
    }
   ],
   "source": [
    "print(data1.shape)\n",
    "print(data2.shape)\n",
    "print(norm_data.shape)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### No Duplicates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data1['sample_id'].values) == len(np.unique(data1['sample_id'].values))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### All gene names being with `_at`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all(s == '_at' for s in [s[-3:] for s in norm_data.index.values])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "class Patient:\n",
    "    def __init__(self, identifier, clinical_group, sample_dict):\n",
    "        '''\n",
    "        Can add more attributes here\n",
    "        '''\n",
    "        assert identifier[:6] == 'MAARS_'\n",
    "        assert isinstance(sample_dict, dict)\n",
    "        assert len(set(sample_dict.values()).intersection({'LES', 'NON_LES'})) > 0\n",
    "        assert clinical_group in ['AD', 'PSO', 'CTRL']\n",
    "        \n",
    "        self.identifier     = identifier\n",
    "        self.clinical_group = clinical_group\n",
    "        self.sample_dict    = sample_dict\n",
    "        self.summary_transcriptomes = {sample: sample_dict[sample] for sample in set(list(norm_data)).intersection(set(sample_dict.keys()))}\n",
    "        \n",
    "    def __str__(self):\n",
    "        part1 = 'Patient_id:\\t\\t\\t{}\\nClinical Group:\\t\\t\\t{}\\nNumber of samples:\\t\\t{}\\n'.format(self.identifier, self.clinical_group, len(self.sample_dict))\n",
    "        part2 = 'Transcriptome summary:\\n' + ''.join(['\\t{}: {}\\n'.format(_id, self.sample_dict[_id]) for _id in self.summary_transcriptomes])\n",
    "        \n",
    "        return part1 + part2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# ad = 0\n",
    "# not_ad = 0\n",
    "\n",
    "list_patients = []\n",
    "\n",
    "for patient_id in np.unique(data1['MAARS_identifier'].values):\n",
    "    patient_data = data1[data1['MAARS_identifier'] == patient_id][['MAARS_identifier', 'clinical_group', 'sample_id', 'lesional']]\n",
    "    assert len(set(patient_data['clinical_group'])) == 1\n",
    "    \n",
    "    sample_dict = {_id: _les for (_id, _les) in list(zip(patient_data['sample_id'], patient_data['lesional']))}\n",
    "    \n",
    "    list_patients.append(Patient(patient_data['MAARS_identifier'].values[0],\n",
    "                         patient_data['clinical_group'].values[0],\n",
    "                         sample_dict))\n",
    "#     if set(patient_data['clinical_group']) == {'AD'}:\n",
    "#         ad += 1\n",
    "#     else:\n",
    "#         not_ad += 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "246"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(data1['MAARS_identifier'].values).difference(set(data2['patient#Identification#MAARS identifier (MAARS_identifier)'].values)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "94"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data2['patient#Identification#MAARS identifier (MAARS_identifier)'].values)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1317"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Patient_id:\t\t\tMAARS_1_002\n",
      "Clinical Group:\t\t\tPSO\n",
      "Number of samples:\t\t4\n",
      "Transcriptome summary:\n",
      "\tMAARS_1_002_04: NON_LES\n",
      "\tMAARS_1_002_03: LES\n",
      "\n",
      "Patient_id:\t\t\tMAARS_1_003\n",
      "Clinical Group:\t\t\tAD\n",
      "Number of samples:\t\t4\n",
      "Transcriptome summary:\n",
      "\tMAARS_1_003_03: LES\n",
      "\tMAARS_1_003_04: NON_LES\n",
      "\n",
      "Patient_id:\t\t\tMAARS_1_004\n",
      "Clinical Group:\t\t\tPSO\n",
      "Number of samples:\t\t4\n",
      "Transcriptome summary:\n",
      "\tMAARS_1_004_04: NON_LES\n",
      "\tMAARS_1_004_03: LES\n",
      "\n",
      "Patient_id:\t\t\tMAARS_1_005\n",
      "Clinical Group:\t\t\tPSO\n",
      "Number of samples:\t\t4\n",
      "Transcriptome summary:\n",
      "\tMAARS_1_005_03: LES\n",
      "\tMAARS_1_005_04: NON_LES\n",
      "\n",
      "Patient_id:\t\t\tMAARS_1_006\n",
      "Clinical Group:\t\t\tPSO\n",
      "Number of samples:\t\t4\n",
      "Transcriptome summary:\n",
      "\tMAARS_1_006_03: LES\n",
      "\tMAARS_1_006_04: NON_LES\n",
      "\n",
      "Patient_id:\t\t\tMAARS_1_007\n",
      "Clinical Group:\t\t\tPSO\n",
      "Number of samples:\t\t4\n",
      "Transcriptome summary:\n",
      "\tMAARS_1_007_04: NON_LES\n",
      "\tMAARS_1_007_03: LES\n",
      "\n",
      "Patient_id:\t\t\tMAARS_1_008\n",
      "Clinical Group:\t\t\tAD\n",
      "Number of samples:\t\t4\n",
      "Transcriptome summary:\n",
      "\tMAARS_1_008_04: NON_LES\n",
      "\tMAARS_1_008_03: LES\n",
      "\n",
      "Patient_id:\t\t\tMAARS_1_009\n",
      "Clinical Group:\t\t\tPSO\n",
      "Number of samples:\t\t4\n",
      "Transcriptome summary:\n",
      "\tMAARS_1_009_04: NON_LES\n",
      "\tMAARS_1_009_03: LES\n",
      "\n",
      "Patient_id:\t\t\tMAARS_1_010\n",
      "Clinical Group:\t\t\tPSO\n",
      "Number of samples:\t\t4\n",
      "Transcriptome summary:\n",
      "\tMAARS_1_010_04: NON_LES\n",
      "\tMAARS_1_010_03: LES\n",
      "\n",
      "Patient_id:\t\t\tMAARS_1_011\n",
      "Clinical Group:\t\t\tCTRL\n",
      "Number of samples:\t\t4\n",
      "Transcriptome summary:\n",
      "\tMAARS_1_011_04: NON_LES\n",
      "\n",
      "Patient_id:\t\t\tMAARS_1_012\n",
      "Clinical Group:\t\t\tCTRL\n",
      "Number of samples:\t\t2\n",
      "Transcriptome summary:\n",
      "\tMAARS_1_012_03: NON_LES\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print_limit = 10\n",
    "\n",
    "for i, patient in enumerate(list_patients):\n",
    "    if i > print_limit:\n",
    "        break\n",
    "    print(patient)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "ml",
   "language": "python",
   "name": "ml"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
